### PR TITLE
Combination issue while trying to change size on a product page

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Catch exception in ProductController, variable should be null if combination isn't found.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-6113
| How to test?  | Follow ticket..

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9390)

### CODE REVIEW

**Comment by iqit-commerce** [comment]
Thir PR does not solve the issue, it is still not possible to select combination. Onlu diffrence  is that that now it does not throw error

File: controllers/front/ProductController.php
Comment (mickaelandrieu) [inline controllers/front/ProductController.php]: Shouldn't be logged at least? I'm not easy to hide an exception.

Comment (PierreRambaud) [reply controllers/front/ProductController.php]: There was no log in 1.7.3 and don't found anything in legacy to add a log without changing many things (the only log function is protected under PrestasShopException)

Comment (mickaelandrieu) [reply controllers/front/ProductController.php]: how about https://github.com/PrestaShop/PrestaShop/blob/develop/classes/PrestaShopLogger.php ?

**Comment by mickaelandrieu** [comment]
> Thir PR does not solve the issue, it is still not possible to select combination. Onlu diffrence is that that now it does not throw error

Yes ... any hint to help us fix this bug?

**Comment by iqit-commerce** [comment]
@mickaelandrieu I explained problem more detailed in latest comment on Forge, http://forge.prestashop.com/browse/BOOM-6113?focusedCommentId=173029&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-173029

File: controllers/front/ProductController.php
Comment (PierreRambaud) [reply controllers/front/ProductController.php]: Hey, Thanks mate ;)

**Comment by PierreRambaud** [comment]
@iqit-commerce @mickaelandrieu Seen with @marionf I reproduce the exact old behavior and fix the bug. We need to define what will be the next behavior and it's not the priority here :/

File: controllers/front/ProductController.php
Comment (mickaelandrieu) [inline controllers/front/ProductController.php]: I'm curious about it, how does this log message render in BO Logs page?

**Comment by iqit-commerce** [comment]
@PierreRambaud No it is not old behavior. In ps 1.7.3 it was possible to choose combination. I created video which show diffrence, On first screen there is combination configuration, On second ps 1.7.3 and then 1.7.4

https://youtu.be/R0AibzDWl7w

Important in combination configuration is that that there are only two combinations like

1. Size S, Color: Green, - default
2. Size M, Color: Blue,


If wee add two more combinatins like this

3.Size S, Color: Blue
4. Size M, Color: Green,

then issue do not occurs

File: controllers/front/ProductController.php
Comment (PierreRambaud) [reply controllers/front/ProductController.php]: Hahaha, too tired today, thinking it was a file logger... I change this now =)

**Comment by PierreRambaud** [comment]
@iqit-commerce Let's have a last try, should be ok now. (My bad I misunderstand something ;) )

**Comment by iqit-commerce** [comment]
@PierreRambaud Yes, I was about to post this solution, we found out this in same time :) now it is like it was in PS 1.7.3.

But there are still problems in some scenarios

A. 

1. Size S, Color: Green, - default
2. Size M, Color: Blue,
3.  Size M, Color: Green,


And third combination is out of stock (Size M, Color: Green,)
 and in product prerernees settins you set "Display unavailable product attributes on the product page" to NO, it will still not find combination (2. Size M, Color: Blue,)

B. 
Customer still  have to change first to M size to see that product is also avaiable in diffrent colour

**Comment by iqit-commerce** [comment]
@PierreRambaud And I think it should be merged in 1.7.4.x and instead of waiting for 1.7.5 :)

**Comment by PierreRambaud** [comment]
Hum, I not able to take a decision about what behavior it should have in this situation :/ 
You need to create a new forge ticket and explain what behavior you want.

For the milestone,  @mickaelandrieu can confirm (I hope) but it will be released in the next major version.

**Comment by iqit-commerce** [comment]
"But there are still problems in some scenarios

A.

Size S, Color: Green, - default
Size M, Color: Blue,
Size M, Color: Green,
And third combination is out of stock (Size M, Color: Green,)
and in product prerernees settins you set "Display unavailable product attributes on the product page" to NO, it will still not find combination (2. Size M, Color: Blue,)"

Scenario A is still and issue and not changig behavior

**Comment by eternoendless** [comment]
> it will be released in the next major version.

Just a quick reminder: 1.7.5 is a _minor_ version. 

@iqit-commerce While we are aware that this is a major issue for some people, we have decided that this bug does not justify a patch release by itself. Please understand that a release demands a lot of resources, both internally and externally, so we want to keep patch releases to a minimum.

**Comment by iqit-commerce** [comment]
If there will be another ps 1.7.4.x before 1.7.5 then I really do not understand how you assign PR yo develop or current 1.7.4
X . As IT was explained to me in one pr, there should be used 1.7.4.x for bugfixes, especially for issues introduced in 1.7.4

Edit. I do not mean to realase path version just for this bugfix, but including this bugfix in 1.7.4.3 IF it will be realeased

**Comment by PierreRambaud** [comment]
@mickaelandrieu @eternoendless Shoult it be reviewed and waiting for QA? Or wait for feedback and need improvement?

**Review by mickaelandrieu — APPROVED** [review]

**Comment by eternoendless** [comment]
@iqit-commerce I understand your confusion, which is totally justified as we have recently changed our rules on what we merge where.

Historically, there was a golden rule: bug fixes go in the patch branch (1.7.4.x) and the rest in develop.

In order to reduce the number of "off-schedule" releases, we have decided that starting now, we'll only accept regressions and critical bug fixes in patch branches. Even though this issue *is* a regression, we have already released too many patch versions for 1.7.4 (including critical bug fixes). Of course, this doesn't mean we _won't_ fix them, quite the opposite. It's just that we'll add those fixes in the next minor version.

This has been documented in this article: [Contribution guidelines - target branch](https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#target-branch).

In the event we release a hypothetical 1.7.4.3, we may cherry-pick this PR.

**Comment by marionf** [comment]
@PierreRambaud I would like to test if your PR works also when you click on the quick view but I can't because of this issue on develop branch: http://forge.prestashop.com/browse/BOOM-6017 :(

**Comment by PierreRambaud** [comment]
@marionf SIR YES SIR! I will fix this one :)

**Comment by PierreRambaud** [comment]
@marionf Waiting for https://github.com/PrestaShop/PrestaShop/pull/9403

**Comment by matks** [comment]
https://github.com/PrestaShop/PrestaShop/pull/9403 has been merged, @marionf you should be able to validate this PR now

**Comment by marionf** [comment]
Need a rebase, no ?

**Comment by marionf** [comment]
All good, thanks @PierreRambaud

**Comment by Tomskor** [comment]
This doesnt work - if I add changes to original files presta 1.7.4.2 only change is dont have any error when I am choosing combination but EVERYTIME load another product so its a bigger problem. 

How to fix it and when will be update or patch?

**Comment by matks** [comment]
@Tomskor Our QA team was able to see this bug fixed so I think the bug fixed by this PR and your bug are 2 different bugs but they produce the same output.

Please create an issue if you have not already created one, the QA team will then to understand the exact conditions in which you observe the bug.

Once we know the required conditions to reproduce your bug we will be able to fix it.

**Comment by Tomskor** [comment]
@matks propably you'r right, we update our custom theme and problem is gone for now and choosing a sizes of clothes works fine, everything chanes when I replace old product.tpl file.

**Comment by ghost** [comment]
Hi guys,

My business shop is based on product combinations, so this feature is basic for us and for many people. Fix this please as soon as possible. Rolling back to 1.7.3.4 is not an option for us.

We have products with a lot of combinations, and with 1.7.3.4 worked like a charm.

Now we have the "An error occurred while processing your request" warning when we select any combi.
 It's not a template problem, or an addon module. With all third parties modules disabled and the Classic theme as default you can find this bug.

Products with 2 combinations or less are not affected. We can't wait for a month for a fix. So please guys fix this.

Thank you very much.

**Comment by mickaelandrieu** [comment]
Hi @AbelEscrig,

if rolling back to 1.7.3.4 is not an option for you, the best you can do is to ask a developer to apply the patch @matks have done here that will be available for 1.7.5.0 only that we plan to release in October (:crossed_fingers: ).

As for now, we don't plan to make a new patch version of 1.7.4.x (to be clear: no 1.7.4.4 planned).

Thanks for your understanding.

**Comment by ghost** [comment]
I'm a newbe, so please can you tell me where is the patch here? It works? It's my first time on GitHub. If it's only modify a line copying/pasting code i can do it myself. Please help me and sorry

**Comment by mickaelandrieu** [comment]
@AbelEscrig sure, you need to report on your shop the update done in this pull request : look at "Changes" tab.

Regards

**Comment by ghost** [comment]
Hi guys,
After copypasting the raw content of the 3 files, the web loads, but the product page not.
I'm going mad... :(

**Comment by mickaelandrieu** [comment]
Hi,

Did you copy all the file content? o_O

If you're not a developer you should not touch the files, find someone able to do it, maybe on the official forum...

**Comment by ghost** [comment]
I've copied all the code from the 3 new files replacing the whole code of the old ones. Is this correct?

Can you contact me in private and help me to do this? I can send you my 1.7.4.2 files

**Comment by ttoine** [comment]
In this patch, most modifications are for 1.7.5 (first file, code clean, second and third files, updating "getIdProductAttributeByIdAttributes")

If you want to apply the modification for 1.7.4.2, you can try to just add [the line 960]( https://github.com/PrestaShop/PrestaShop/pull/9390/files#diff-514c245ae9a59a3ffd37a1e35d97f00cR960) in "controllers/front/ProductController.php". You should have something like that:

```
if (!empty($groups)) {
    $requestedIdProductAttribute = (int) Product::getIdProductAttributesByIdAttributes(
        $this->product->id,
        $groups,
        true
    );
}
```

Take great care:

 * Use only this specific modification, don't try to replace the entire files (other modifications might have been done before for 1.7.5)
 * Test it before applying to your production shop, e.g on a local or staging installation
 * If this is critical for your business and you don't have the required skills, please consider to hire a freelance or an agency to get help

It should work with the classic theme. It has been reported in this discussion by @Tomskor that [a similar issue might be related to a custom theme](https://github.com/PrestaShop/PrestaShop/pull/9390#issuecomment-415661929).

**Comment by ghost** [comment]
Hi @ttoine ,
First thank you for your help. If only add the 960 line to the 1.7.4.2 and of course forgetting the other files, i've got a blank white page. Using Classic Theme and Custom theme. It's not working.

The same thing happens when i replace the 3 new files.

Please i need help. I can't wait for October. I have my business blocked. In my honest opinión, Prestashop never should be bugged in something important such as the combinations.

**Comment by ttoine** [comment]
As I told you, if this is critical for your business, you must hire someone with skills. No need for a full time job. Many freelancers and agencies can do it. Buy them a day or two of service. And when they found a solution, tell them to share that solution here.

**Comment by ghost** [comment]
In my honest opinion:

1. I would never share nothing what i have to pay for, to fix something that is free.
2. Every Prestashop update breaks important things bringing useless new features that had to remake a lot of things.
3. The stable 1.4.X never would be released with bugs like this.
4. You suggest changes that simply doesn't work in a Classic theme with all non native modules off.
5. Thank you for your help. I don't wat to be rude, but Prestashop always was fun. Now is a completely nightmare. Most of the users have paid a lot of money in modules. So if we want to amortize the investment we are hook up. Is the reason to stay.
6. I read here a developer saying that these kind of things doen' justify an update. Multiple combinations broken is not an important thing? Come on guys. It's a joke.

**Comment by ttoine** [comment]
1. PrestaShop is not free as in free beer, but as in free speech. PrestaShop is an open source project, not a freeware.
2. Before doing any migration on a production server critical for your business, the best practice is to test locally, or better, on a testing server with the exact same configuration than your production. When your tests are ok, you do a back up so you can roll back to the previous version, just in case. And then you migrate. It also means that you should open a document to list all the tests to do.
3. The 1.7.4.x introduced some issues, but also solved a lot of issues. It also includes some refactoring from the legacy code. We are looking for more users from the community to test the beta and report bugs. This way bugs are discovered and fixed before the release. Your help is welcome for the next release.
4. I don't want to be rude neither. But please keep in mind that when you use any business solution, open source or not, you must consider to get support from professionals. I am quite sure that you pay your bank, your accountant, your domain name and your hosting provider. This is the same for servers and software maintenance when skills are required. The reason why modules are not cheap is that their developers have to pay their bills at the end of the month, like you. And you buy them to get features that you need for your business. A lot of merchants hire freelancers or agencies to maintain their shop and apply patches, because this is critical for their business.
5. The release cycle is currently every 3 or 4 months. Most open source project, even backed by a company, have similar or even longer development cycles.

Again, if you don't have the skills to apply a patch, the best way is find someone to do it for you.

**Comment by ghost** [comment]
You're right is some things but not in everything.

No problem being a tester as a user i am, but when you release a software it has to be tested in BASIC things. Agree with you in that sometimes some things have to be patched after a stable release. But when a simple combination doesn't work is due to developers don't test those basic things before.

When a car mechanic works on a engine, it's not the user who has to test the security, is the mechanic brfore delivering the car. Do i have to pay to test something that can damage my business? No.

This bug compromises the full shop, and the answer here is, okay maybe in October in 1.7.5.0 will be fixed. Are you sure of that?

I spent 2 weeks waiting for a fix from the custom theme developer to adapt the 1.7.3.4 custom theme to 1.7.4.2. 

My business has spent too much money on this open source software during the last 10 years, and always is the same history. The jump from 1.4.5.2 to 1.5 was a nightmare. Now the 1.6 users has serious problems to update to 1.7, and a simple update to 1.7.4 broke the shop. It could be avoided.

This is a platform where a lot of business have customers critical data. Many amateurs are breaking Prestashop in every update, and they are the same people that after that, most of the times are selling buggy addons in prestashop addons page with pro prices.

Open code YES, free access to amateurs to modify it NO ,thank you. GTA V mod community is good for that purpose, not Prestashop with millions of dollars/euros floating and millions of users that have trusted on your knowledge.

No problem to pay someone for a job well done but with warranty. The problem is that every month i have to pay someone to fix stupid things, and in more than a month, nobody on this thread has idea of how to fix this problem. I will not pay to the same people that is causing the problem. What a business! You create the problem, you get the money to fix it? It's a joke.

Nothing personal with you @ttoine, believe me. This post is for those developers who hide when the tide goes up. You are trying to helping me and thank you for that.

Many users here think the same as me, i'm sure of that.

Conclusión: Don't release as a stable, something that is not stable is very basic things. People will upgrade trusting in stable version. This is a waste of time.

**Comment by mickaelandrieu** [comment]
> This post is for those developers who hide when the tide goes up.

As a member of the Core team, I feel I may be concerned, do I?

We're not hiding, the project is open source and all the issues were just migrated to GitHub so everyone can see what happens here, your own comments included (you're free to remove them, but still).

Once the issue was found by a user, the issue was fixed: so you can consider at least 3 time period for a bug to be fixed.

1) the bug needs to be known: here we have invested a lot, but like A LOT in our test suite. We have a dedicated team for the quality (between 3 to 5 members that we pay of course). If you don't believe me, you can take a look at the [test suite build](https://travis-ci.org/PrestaShop/PrestaShop/builds/410242769?utm_source=github_status&utm_medium=notification), which is open, too. But we can't test everything and sometimes bugs may happen, this is why we release beta versions.

2) the current reality, is that almost no one tries the beta versions: after all, why should you care about it? Let's wait for the stable version! And once we have fixed all bugs _the beta testers have found,_ we release the stable version.

3) then people upgrade their shop and start to complain: "wtf this is a basic feature, why *** (chose the issue you want) doesn't work? So we start to work on patch versions, fixing all the bugs found by the users. For 1.7.4.x release, we have published 2 releases, 2 more releases is a big work for a little team of developers: you can believe me on that.

4) but eh! We have to work on the next minor release, too. So after 2 patch versions, we have decided that every bug will be fixed in `develop` branch for the next `1.7.5.0`. We are already at the 4th step of this process and all this work was provided to you for free (as a beer and as a speech).

5) And now, let's welcome you! Blaming us for a bug **already fixed**, insulting us of "amateurs" when you're unable to apply a simple patch on your shop, refusing even the "idea" to contribute for free to a project that make you eat for maybe a long time! You should be humble and thankful, not only for the money PrestaShop (as a company) invest to provide the project you use for free but also to **every contributor** (paid or not paid: ~ 10%) that make this project still used, still maintained and still improved every week.

6) And for what it means, we don't do support on GitHub: at least no one is paid for that by PrestaShop because we sell support. Everyone that helped you here have done it to help you, for free or in their free time: this is how people should behave in an open source community. Think about it.

Regards

**Comment by ghost** [comment]
What are you talking about man?

1. Your fix simply doesn't work, as many users communicated here. DON'T work at all. They simply doesn't fight anymore. That's all.

2. The stable 1.7.4.2 contains this bug, so wait for a stable version? Your team or anybody out there tagged this bugged version as a stable.

3. I'm capable to copy paste code, and replace files. You posted how to do it, But sometimes miracles doesn't work with software. It's code. The problem is that the guy who broke the software doesn't have any idea of what he did it.

4. I can demostrate you and others that this community is plenty of amateurs billing pro invoices. I made a lot of complains to Prestashop team due to this. There is lot of fake modules for more than 100 bucks that you have to modify to implement them, and nobody talked about that in the description before pay. A module is for install and 0 problems You pay for that. Prestashop can do that. Is another debate, but there are too many "developers" without any quality control from the Prestashop team. And people is buying trusting that all the content in the addons page is tested. If you have been ofended sorry, but it's the true.

6. You have job because people like me are interested in this e-commerce and others. If these things continue like this, people drop the idea to choose Prestashop, so maybe you change your point of view.

7. If I have to pay to talk with you to tell me how to put a file on a server. You will never have happy customers in a near future. I have 3 enterprises (one with pretashop) and never, NEVER i would say to a customer to pay to change a file. Step down the cloud and please fix a buggy beta version that never should be released as a stable.

8. Now you can ban me if you feel better, but don't insult my intelligence and the intelligence of other users that are tired of buggy versions, and expensive nightmares everytime somebody has the wrong idea to improve something that works like a charm. Think about it.

9. if you feel alluded as an amateur, it's you problem not mine, but please do not use that rhetoric to offend me. You can't.

**Comment by mickaelandrieu** [comment]
Hi,

first of all, I apologize if my remarks may have been offending to you. We do not tolerate insults, so my behavior was not appropriate: sorry!

Then the bug was fixed. Validated by 2 people in QA and by a [member of the community](https://github.com/PrestaShop/PrestaShop/pull/9390#issuecomment-415661929): either you have found another bug or the patch has been incorrectly applied.

Honestly, your behavior does not help me help you. For my part, I did my best and I leave the conversation open in case someone wants to help you.

Good night

**Comment by ghost** [comment]
First test: Replaced the 3 files suggested by @ttoine in the correct place. Blank product page forcing compilation and without compilation ----> No solution

Second test: replace the line 960 of the appropiate file suggested. Blank product page forcing compilation and without compilation ----> No solution


Products here with more than 10 possible atributes with more than 10 variables in every attribute. Working like a charm in 1.7.3.4

"An error occurred while processing your request" warning every time i try to choose any combi in fresh 1.7.4.2 with Classic theme and Custom theme (no native modules disabled)

No problems in 1.7.4.2 with only 2 atributes with 2 variables each one

**Comment by ttoine** [comment]
I think that your example of the car speaks by itself. You don't understand the very nature of PrestaShop.

> When a car mechanic works on a engine, it's not the user who has to test the security, is the mechanic brfore delivering the car

A car is a product, engineered and distributed by a manufacturer with garanties. You need a driving license to use it. You have to buy it, it's not available for free. And when you pay a mechanic, it's because you don't have the skills to fix or maintain something, with all the tests and security.

PrestaShop is not a product. It's an open source project. You are free to download it, without any requirement: it's up to you to be skilled enough to use it. So it means that anyone can build the shop of their dream, using the project as a solid base. There is an infinite combinations of server, operating system, PrestaShop version, installed modules, themes, and customisations. It is not possible (and it will never be possible) for the small team in the company to test every possible combinations. For this, the project rely on the community of users and contributors. Like any open source project, btw.

And, of course, because you built it you own way, it's also your task to upgrade it and maintain it. At your own risk, or, you can hire a skilled "mechanic" to do that maintenance and tech work for you, like for your can. I hope the comparison is clear?

Also, please keep in mind that PrestaShop as a lot of legacy code and technos that we are improving and refactoring at each release. Automated tests are added when / where it is possible. That's why this kind of issue are possible in a new version: rewriting some part of the code might have an effect on an other part, and that is not easy to detect. That is an other reason why we rely on the community of users and contributors to help us with their tests, and then to report those issues.

So again, rules before upgrading to a new version:
 - you should never apply an update on production before testing it
 - test on another installation of PrestaShop than the one you are using for production
 - write a checklist of things important to test: your modules, your theme, and main features you are using, and follow it on your testing installation
 - backup your production installation (files and database) to be able to roll back

Quoting Mickael:

> either you have found another bug or the patch has been incorrectly applied.

I agree with him, you are certainly in one of this case. I hope you understand now that it's up to you to apply the patch, like other people did in this discussion. Or to hire someone skilled.

About [this answer](https://github.com/PrestaShop/PrestaShop/pull/9390#issuecomment-417061359):

1. the fix worked for two other guys who knows how to apply a patch. period.
2. you don't have any idea of the work to release a new version of PrestaShop, and everybody here tells that applying the patch is easy
3. the guy who fixed the issue did it for the develop branch, so for the upcoming 1.7.5, and not for 1.7.4.x; and he knows well what he is doing, he is not an "amateur"
4. could you please contact the Addons support and report to them the modules with this kind of issues? that would be helpful and GitHub is not the place for that.
5. you can build your shop with PrestaShop because PrestaShop hired developers to develop it; and then, developers continue to have jobs in order to answer the needs of the growing users base; how could we turn that into a virtuous circle, instead of being disrespectful?
6. again, PrestaShop is an open source project and is not responsible for your issues, maybe you should read the license; if you need technical support, there are plenty of skilled people you can hire everywhere in the world
7-8. the one who is looking for respect should always start with being respectful; this was not respectful to speak about the developers as being amateurs; nobody told you that you are an amateur as a merchant, however, you are not a skilled developer, or you would know how to apply a patch, and maybe check if you might have another issue.

That said it's midnight, and I am answering on my free time. Please calm down on this discussion.

**Comment by ghost** [comment]
Definitely @ttoine, you're treating me as an idiot. So please stop or i'll start to say what i really think about you from my personal point of view.

The fact that 2 guys said that they fixed the issue only means that THEIR issue can be patched. This doesn't means that mine issue was the same issue. So before start repeating like a bird that you have the solution and induce that the rest of the humanity are morons, please start to thinking that maybe there's another big issue with similar warning message. It's common sense.

But not, you prefer to say that the problem is that i'm not capable to apply a patch. From the moment you are treating users as fools, you @ttoine, become an amateur, and doesn't matter how many knowledge you have in your brain my friend. Learn to be polite. I was always polite until you started with this sh*t and if someone can't find a solution, don't  treat people as idiots. Get out of the office, take a Coke, go to the street and improve how you treat people who do not understand you, or who simply do not agree with what you propose. Don't give me ethics masterclass. You are wasting my time.

We are users, and as users, we invest time to test your software EVERY WEEK. OUR TIME IS MONEY TOO. The fact that this is open source doesn't justify that only your efforts are worthy of being paid. I'm not an slave to test your beta experiments every fu**king week when a non stable version that never should be released was realeased as stable.

From the moment that this 'free' open source is filling your buckets directly or indirectly, you have a minimum of responsability, especially when someone modifies something that worked years ago. Prestashop team is the responsible to break the code that weeks ago worked like a charm, and now it can't do more than 3 combinations by itself, mother of god.

Read the f***ing thread more than once and you'll find that i was calmed before people like you when something doesn't work, treat people as idiots. Of course, in this world there are people who are treated like idiots in their faces every day in their lifes and do not realize it. But that is not my problem.

From the moment that this guy wrote the patch for a 1.7.5.0 scenario, this is not applicable for a 1.7.4 as i explained, so if i have to do changes more than copy paste, you should to explain users what to do and not  leading me to copy paste files as you did, when YOU KNEW THAT MAYBE THIS DOESN'T WORK and is needed to do more things.

Please stop contacting me and bothering me, or i will report. I don't want this kind of support.

**Comment by ttoine** [comment]
> they fixed the issue only means that THEIR issue can be patched
This is what we try to tell you: you might have an other issue, so please help us to investigate instead of insulting us... I don't treat you as an idiot, you have just the feeling that I do. I and others tried to help you on our free time, that doesn't give you the right to tell we are amateurs.

> You are wasting my time.
Actually, you are wasting your time, and the time of the developers. If you could provide more information, we could certainly help you. But, instead, you are just insulting us.

> slave
You are not a slave, but, you built a custom shop, and nobody else than you can test it (that's the rule with most open source projects). I am glad to now that you test every week. So now, open a bug about your issue, and answer to the team with the details. Get help on the forum and search if other than you have the exact same problem.

Last call, stop to use bad vocabulary like sh*t and fu**king. Fact: I am polite, you are not.

BTW, if you want to stop to get notifications from this Pull Request discussion or from the PrestaShop project, please change your settings in your GitHub profile. Otherwise next time someone adds a comment, you will be notified.

**Comment by ghost** [comment]
1. The defintition of polite is very wide, and include something more than bad words.

2. Don't cut text altering the context. The users that read this thread are not stupid.

3. A fresh installation of Prestashop 1.7.4 with no modules and more than 3 combinations simply crashes on product page. It's not my "built shop". Do your homework. Mine is done.

4. I came here searching solutions, but i found people saying that the problem is on the user side because of we don't know how to apply a patch. You do not know who I am or my knowledge, but you put them in doubt from the moment your solution does not work. That defines you.

5. You ego prevents you from seeing what is really affecting the users and their business. This is not a game. I have a 1.7.3.4 working because WE MAKE backups and tests before. But the complaint here is that you release bugged versions as stable versions

6. You don't have any interest in to find what is happening really. Only repeat that we don't have to apply an useless patch in this case. So yes, at this moment, when you insult my intelligence, i treat you at the same level.

Stop bothering me, do your job if you have one. But i will speak, say whatever i consider. Give solutions and listen to the users before getting them angry, but don't say what do i have to think or to do. Where i come from we have freedom of expression.

**Comment by ttoine** [comment]
Freedom of expression does NOT give the right to use bad vocabulary or insults. It's all about respect. And I dont' insult you. You do.

Again, please update your GitHub settings to stop to get notifications.

**Comment by eternoendless** [comment]
In order to avoid [Godwin's Law](https://en.wikipedia.org/wiki/Godwin%27s_law), I'm closing this PR to further comments. 

This is not a discussion forum, and Pull Requests are not the right place to ask for help. If you think you have found a bug, and if it's not fixed by this PR, please [open an issue](https://github.com/PrestaShop/PrestaShop/issues/new/choose). If you need help with your shop, you can ask for help in the [Community Forums](https://www.prestashop.com/forums/).

**Pull request merged**
